src/python/pants/engine/native.py

```diff
@@ -589,14 +599,14 @@ def binary(self):
   @memoized_property
   def lib(self):
     """Load and return the native engine module."""
-    return self.ffi.dlopen(self.binary)
+    lib = self.ffi.dlopen(self.binary)
+    _FFISpecification(self.ffi, lib).register_cffi_externs()
+    return lib
 
   @memoized_property
   def ffi(self):
     """A CompiledCFFI handle as imported from the native engine python module."""
-    ffi = getattr(self._ffi_module, 'ffi')
-    _FFISpecification(ffi).register_cffi_externs()
-    return ffi
+    return getattr(self._ffi_module, 'ffi')
```

Comment:
Considering that `register_cffi_externs()` modifies the `ffi` object and not the `lib` object, I would think we would want to keep that call in this method to avoid spooky mutation of the `ffi` object happening whenever we first happen to access the `lib` object.

Reply:
Unfortunately, constructing the `_FFISpecification` inside of `self.ffi` triggered infinite recursion when I attempted to feed it `self.lib` (which required `self.ffi`, etc, etc).